### PR TITLE
En-2.16.46

### PR DESCRIPTION
**Pull request by Nero-Hu**
Pls ignore the branch name. It's a typo.
It should be EN-2.16.46.

### CODE REVIEW

File: iOS-EN（裁剪版）/headers/Agora Interactive Whiteboard Objective-C Overview-template.md [hunk 1]
```diff
@@ -133,6 +134,7 @@ Agora provides interactive whiteboard with high reliability and rich features th
 | Method                                                       | Description                          |
 | :----------------------------------------------------------- | :------------------------------ |
 | [debugInfo]([WhiteRoom debugInfo:])  | Gets debug logs. |
+| [requestSlideLogToFilePath]([WhiteSDK requestSlideLogToFilePath:result:])  | Gets debug logs. |
```
Comment (mst3k4ever) [inline iOS-EN（裁剪版）/headers/Agora Interactive Whiteboard Objective-C Overview-template.md]: Edit: "the debug"

File: iOS-EN（裁剪版）/headers/Displayer/WhiteCommonCallbacks.h [hunk 2]
```diff
@@ -94,6 +94,26 @@ NS_ASSUME_NONNULL_BEGIN
  */
 - (void)logger:(NSDictionary *)dict;
 
+/**
+ Reports that the SDK tries to recover. 
+ 
+ Called when the SDK crashes due to insufficient memory.
+
+ Currently, only real-time rooms can trigger this callback. This callback is triggered up to three times in one crash.
```
Comment (mst3k4ever) [inline iOS-EN（裁剪版）/headers/Displayer/WhiteCommonCallbacks.h]: Edit: "This callback is triggered a maximum of three times per crash."

File: iOS-EN（裁剪版）/headers/Displayer/WhiteCommonCallbacks.h [hunk 3]
```diff
@@ -94,6 +94,26 @@ NS_ASSUME_NONNULL_BEGIN
...
+ */
+- (void)startRecoveringFromMemoryIssues;
+
+/**
+ Reports that the SDK finishes one recovery.
```
Comment (mst3k4ever) [inline iOS-EN（裁剪版）/headers/Displayer/WhiteCommonCallbacks.h]: Edit: "finishes a recovery."

File: iOS-EN（裁剪版）/headers/Displayer/WhiteCommonCallbacks.h [hunk 4]
```diff
@@ -94,6 +94,26 @@ NS_ASSUME_NONNULL_BEGIN
...
+ Currently, only real-time rooms can trigger this callback. This callback is triggered up to three times in one crash.
+ */
+- (void)startRecoveringFromMemoryIssues;
+
+/**
+ Reports that the SDK finishes one recovery. 
+ 
+ Called when the SDK crashes due to insufficient memory and then finishes one recovery.
+  
+ If the recovery fails, the SDK will not automatically retry.
```
Comment (mst3k4ever) [inline iOS-EN（裁剪版）/headers/Displayer/WhiteCommonCallbacks.h]: Edit: "SDK does not automatically"

File: iOS-EN（裁剪版）/headers/Displayer/WhiteDisplayer.h [hunk 5]
```diff
@@ -223,6 +223,14 @@ NS_ASSUME_NONNULL_BEGIN
  */
 - (void)getSceneSnapshotImage:(NSString *)scenePath completion:(void (^)(UIImage * _Nullable image))completionHandler;
 
+
+/**
+   Gets the screenshot of the specified scene.
+ 
+   **NOTE:** The size of the screenshot equals to that of the current whiteboard view.
```
Comment (mst3k4ever) [inline iOS-EN（裁剪版）/headers/Displayer/WhiteDisplayer.h]: Edit: "screenshot is equal to that of"

File: iOS-EN（裁剪版）/headers/Object/WhiteMemberState.h [hunk 6]
```diff
@@ -18,30 +18,23 @@ typedef NSString * WhiteApplianceNameKey NS_STRING_ENUM;
...
- Rectangle.
- */
+/** Rectangle. */
 extern WhiteApplianceNameKey const ApplianceRectangle;
+/** Eraser. */
+extern WhiteApplianceNameKey const ApplianceEraser;
 /**
- Eraser.
+ * A pencil eraser used to erase any part of the pencil strokes.
+ * This tool only applies for NewPencil. You need to set disableNewPencil to false before using the pencil eraser.
```
Comment (mst3k4ever) [inline iOS-EN（裁剪版）/headers/Object/WhiteMemberState.h:35]: Code font for "disableNewPencil" and "false"?

File: iOS-EN（裁剪版）/headers/Object/WhiteMemberState.h [hunk 7]
```diff
@@ -96,4 +89,19 @@ extern WhiteApplianceShapeTypeKey const ApplianceShapeTypeSpeechBalloon;
 @property (nonatomic, strong, readwrite) NSNumber *strokeWidth;
 /** The font size of the text. */
 @property (nonatomic, strong, readwrite) NSNumber *textSize;
+/**
+ Whethter to allow directly selecting and editing whiteboard texts:
```
Comment (mst3k4ever) [inline iOS-EN（裁剪版）/headers/Object/WhiteMemberState.h]: Edit: "Whether" (typo)
Edit: "text:"

File: iOS-EN（裁剪版）/headers/Object/WhiteMemberState.h [hunk 8]
```diff
@@ -96,4 +89,19 @@ extern WhiteApplianceShapeTypeKey const ApplianceShapeTypeSpeechBalloon;
...
+ 
+ - `YES`: Allow directly selecting and editing whiteboard texts.
```
Comment (mst3k4ever) [inline iOS-EN（裁剪版）/headers/Object/WhiteMemberState.h]: Edit: "text."

File: iOS-EN（裁剪版）/headers/Object/WhiteMemberState.h [hunk 9]
```diff
@@ -96,4 +89,19 @@ extern WhiteApplianceShapeTypeKey const ApplianceShapeTypeSpeechBalloon;
...
+ - `NO`: (Default) Do not allow directly selecting and editing whiteboard texts.
```
Comment (mst3k4ever) [inline iOS-EN（裁剪版）/headers/Object/WhiteMemberState.h]: Edit: "text."

File: iOS-EN（裁剪版）/headers/Object/WhiteMemberState.h [hunk 10]
```diff
@@ -96,4 +89,19 @@ extern WhiteApplianceShapeTypeKey const ApplianceShapeTypeSpeechBalloon;
...
+ */
+@property (nonatomic, assign, readwrite, nullable) NSNumber *textCanSelectText;
+/**
+ Whether to support draw dotted lines:
```
Comment (mst3k4ever) [inline iOS-EN（裁剪版）/headers/Object/WhiteMemberState.h]: Edit: "drawing" (or "the dotted line feature")

File: iOS-EN（裁剪版）/headers/Object/WhiteMemberState.h [hunk 11]
```diff
@@ -96,4 +89,19 @@ extern WhiteApplianceShapeTypeKey const ApplianceShapeTypeSpeechBalloon;
...
+ You need to set disableNewPencil to false before joining the channel.
+
+ - `YES`: Support draw dotted lines.
```
Comment (mst3k4ever) [inline iOS-EN（裁剪版）/headers/Object/WhiteMemberState.h]: Edit: "drawing" (or "the dotted line feature")

File: iOS-EN（裁剪版）/headers/Object/WhiteMemberState.h [hunk 12]
```diff
@@ -96,4 +89,19 @@ extern WhiteApplianceShapeTypeKey const ApplianceShapeTypeSpeechBalloon;
...
+ - `NO`: (Default) Do not support draw dotted lines.
```
Comment (mst3k4ever) [inline iOS-EN（裁剪版）/headers/Object/WhiteMemberState.h]: Edit: "drawing" (or "the dotted line feature")

File: iOS-EN（裁剪版）/headers/Object/WhiteMemberState.h [hunk 13]
```diff
@@ -96,4 +89,19 @@ extern WhiteApplianceShapeTypeKey const ApplianceShapeTypeSpeechBalloon;
...
+/**
+ Whethter to allow directly selecting and editing whiteboard texts:
+ 
+ - `YES`: Allow directly selecting and editing whiteboard texts.
+ - `NO`: (Default) Do not allow directly selecting and editing whiteboard texts.
+ */
+@property (nonatomic, assign, readwrite, nullable) NSNumber *textCanSelectText;
+/**
+ Whether to support draw dotted lines:
+ You need to set disableNewPencil to false before joining the channel.
```
Comment (mst3k4ever) [inline iOS-EN（裁剪版）/headers/Object/WhiteMemberState.h]: Code font for "disableNewPencil" and "false"?

File: iOS-EN（裁剪版）/headers/Room/WhiteRoomConfig.h [hunk 14]
```diff
@@ -111,6 +111,14 @@ NS_ASSUME_NONNULL_BEGIN
  */
 @property (nonatomic, assign) BOOL disableEraseImage;
 
+/**
+ Whether to enable floatbar.
```
Comment (mst3k4ever) [inline iOS-EN（裁剪版）/headers/Room/WhiteRoomConfig.h]: Edit: "the floatbar."

File: iOS-EN（裁剪版）/headers/Room/WhiteRoomConfig.h [hunk 15]
```diff
@@ -111,6 +111,14 @@ NS_ASSUME_NONNULL_BEGIN
...
+
+ - `YES`: Enable floatbar.
```
Comment (mst3k4ever) [inline iOS-EN（裁剪版）/headers/Room/WhiteRoomConfig.h]: Edit: "the floatbar."

File: iOS-EN（裁剪版）/headers/Room/WhiteRoomConfig.h [hunk 16]
```diff
@@ -111,6 +111,14 @@ NS_ASSUME_NONNULL_BEGIN
...
+ - `NO`: (Default) Disable floatbar.
```
Comment (mst3k4ever) [inline iOS-EN（裁剪版）/headers/Room/WhiteRoomConfig.h]: Edit: "the floatbar."

File: iOS-EN（裁剪版）/headers/SDK/WhiteSDK.h [hunk 17]
```diff
@@ -137,6 +137,13 @@ NS_ASSUME_NONNULL_BEGIN
  */
 - (void)setCommonCallbackDelegate:(nullable id<WhiteCommonCallbackDelegate>)callbackDelegate;
 
+/**
+ Writes the slide log into a specified file.
+ 
+ @param path The path to the specified file. If the file does not exist, it will be created. If the file already exists, the slide log will be written at the end of the file.
```
Comment (mst3k4ever) [inline iOS-EN（裁剪版）/headers/SDK/WhiteSDK.h]: Edit: "it is created." (avoid future tense)
Edit: "is written" (or "is appended to the end of the file.")

**Review by mst3k4ever — COMMENTED** [review]
:)

File: iOS-EN（裁剪版）/headers/SDK/WhiteSDK.h (hunk 17 above)
Comment (Nero-Hu) [reply iOS-EN（裁剪版）/headers/SDK/WhiteSDK.h]: Wow, I like "append to". thx~

**Pull request merged**